app/controllers/session_controller.rb

```diff
@@ -332,6 +333,34 @@ def create
     end
   end
 
+  def passkey_challenge
+    render json: DiscourseWebauthn.stage_challenge(current_user, secure_session)
```

Comment:
Should we raise a `Discourse::NotFound` here too if `SiteSetting.experimental_passkeys` is not `true`?

Reply:
We can but I don't think it matters too much... this endpoint creates a challenge (random string), stores it in the session and returns it.